discovery-infra/day2.py

```diff
@@ -0,0 +1,106 @@
+import os
+import utils
+import waiting
+import uuid
+import assisted_service_api
+import consts
+
+
+def execute_day2_flow(internal_cluster_name, num_workers, api_vip_ip, api_vip_dnsname):
+    utils.recreate_folder(consts.IMAGE_FOLDER, force_recreate=False)
+    client = assisted_service_api.create_client(
+            url=utils.get_assisted_service_url_by_args(args=args)
+            )
+    cluster_id = str(uuid.uuid4())
+    random_postfix = cluster_id[:8]
+    ui_cluster_name = internal_cluster_name + f'-{random_postfix}'
+    cluster = client.create_day2_cluster(
+            ui_cluster_name, cluster_id, **_day2_cluster_create_params()
+            )
+
+    set_cluster_pull_secret(client, cluster_id, args.pull_secret)
+
+    image_path = os.path.join(
+            consts.IMAGE_FOLDER,
+            f'{args.namespace}-installer-image.iso'
+        )
+    client.generate_and_download_image(
+            cluster_id=cluster.id,
+            image_path=image_path,
+            ssh_key=args.ssh_key,
+            )
+
+    day2_nodes_flow(client, internal_cluster_name, cluster, image_path, num_workers, api_vip_ip, api_vip_dnsname)
+
+
+def day2_nodes_flow(client, cluster_name, cluster, image_path, num_worker_nodes, api_vip_ip, api_vip_dnsname):
+    tf_network_name, total_num_nodes = apply_day2_tf_configuration(cluster_name, num_worker_nodes, api_vip_ip, api_vip_dnsname)
+    utils.run_command(f'make _apply_terraform CLUSTER_NAME={cluster_name}')
+    time.sleep(5)
```

Comment:
again why sleep? `utils.run_command()` is blocking, if that's why you added sleep here

Reply:
lets have no sleeps and move them to waits